source/code/providers/Container_HostInventory_Class_Provider.cpp

```diff
@@ -0,0 +1,212 @@
+/* @migen@ */
+#include <MI.h>
+#include "Container_HostInventory_Class_Provider.h"
+
+#include <vector>
+#include <string>
+#include <syslog.h>
+#include <stdio.h>
+#include <uuid/uuid.h>
+
+#include "../cjson/cJSON.h"
+#include "../dockerapi/DockerRemoteApi.h"
+#include "../dockerapi/DockerRestHelper.h"
+
+using namespace std;
+
+MI_BEGIN_NAMESPACE
+
+class Guid
+{
+public:
+    ///
+    /// Create a guid and represent it as string
+    ///
+    static string NewToString()
+    {
+        uuid_t uuid;
+        uuid_generate_random(uuid);
+        char s[37];
+        uuid_unparse(uuid, s);
+        return string(s);
+    }
+};
+
+class ContainerHostInventoryQuery
+{
+public:
+    ///
+    /// Get logs for all containers on the host
+    ///
+    /// \returns Object representing the host
+    ///
+    static Container_HostInventory_Class InspectHost()
+    {
+        Container_HostInventory_Class hostInventoryInstance;
+
+        // Request containers
+        vector<string> request(1, DockerRestHelper::restDockerInfo());
+        vector<cJSON*> response = getResponse(request);
+
+        // See https://docs.docker.com/engine/api/v1.27/#section/Versioning for example output
+        if (!response.empty() && response[0])
+        {   
+            printf("inspect host \n");
```

Comment:
Is this supposed to be a debug message? If so, why not use SCX logging, which I think you pretty much have anyway due to use of the PAL, which I think you'll need for unit test support.

I haven't gotten this far yet, but you do have unit tests for this, right?